packages/app/src/systems/Network/pages/AddNetwork/AddNetwork.tsx

```diff
@@ -1,11 +1,42 @@
 import { Box, Button, Dialog, Focus, Icon, IconButton } from '@fuel-ui/react';
+import { useEffect } from 'react';
 
 import type { NetworkFormValues } from '~/systems/Network';
-import { NetworkForm, useNetworks, useNetworkForm } from '~/systems/Network';
+import {
+  NetworkForm,
+  useNetworks,
+  useNetworkForm,
+  useChainInfo,
+} from '~/systems/Network';
 
 export function AddNetwork() {
   const form = useNetworkForm();
+  const { isDirty, invalid } = form.getFieldState('url');
+  const isValidUrl = isDirty && !invalid;
   const { handlers, isLoading } = useNetworks();
+  const {
+    chainInfo,
+    error: chainInfoError,
+    isLoading: isLoadingChainInfo,
+  } = useChainInfo(isValidUrl ? form.getValues('url') : undefined);
+
+  useEffect(() => {
+    if (isValidUrl && !isLoadingChainInfo && chainInfo) {
+      form.setValue('name', chainInfo.name, { shouldValidate: true });
+    }
+  }, [chainInfo, isLoadingChainInfo, isValidUrl]);
+
+  useEffect(() => {
+    if (chainInfoError) {
+      form.setError('url', {
+        type: 'manual',
+        message:
+          chainInfoError === 'Network request failed'
+            ? 'Invalid network'
+            : 'Unsupported network',
+      });
+    }
+  }, [chainInfoError]);
```

Comment:
We can move this logic to the react-form component it support url validations and also async validations.